### ISSUE TITLE
dot conflicts with graphviz dot program

### ISSUE THREAD

**Issue by dsisnero** [ubnt-intrepid/dot#14]
dot is a very widely used command already for the graphviz program

**Comment by dataf3l** [comment]
I also came here to say the exact same thing. I love that there are new programs and new creativity, please let's be creative and invent a new name, so there is no naming conflict. unfortunately, I have no suggestions, what about "dotfiles" ?

**Comment by marcfrederick** [comment]
There's already a dotfile manager named [dotfiles](https://github.com/rhysd/dotfiles).
Maybe something like dotman (for dotfile manager) could work?

**Comment by daveloyall** [comment]
Agreed.  https://en.wikipedia.org/wiki/Graphviz is installed on all my machines.  The binary is generally `/usr/bin/dot`.

Could anything destructive happen, theoretically, if somebody tried to invoke your `dot` with command line options intended for the graphviz `dot`?

For your convenience:

    Usage: dot [-Vv?] [-(GNE)name=val] [-(KTlso)<val>] <dot files>
    (additional options for neato)    [-x] [-n<v>]
    (additional options for fdp)      [-L(gO)] [-L(nUCT)<val>]
    (additional options for memtest)  [-m<v>]
    (additional options for config)  [-cv]
    
     -V          - Print version and exit
     -v          - Enable verbose mode
     -Gname=val  - Set graph attribute 'name' to 'val'
     -Nname=val  - Set node attribute 'name' to 'val'
     -Ename=val  - Set edge attribute 'name' to 'val'
     -Tv         - Set output format to 'v'
     -Kv         - Set layout engine to 'v' (overrides default based on command name)
     -lv         - Use external library 'v'
     -ofile      - Write output to 'file'
     -O          - Automatically generate an output filename based on the input filename with a .'format' appended. (Causes all -ofile options to be ignored.)
     -P          - Internally generate a graph of the current plugins.
     -q[l]       - Set level of message suppression (=1)
     -s[v]       - Scale input by 'v' (=72)
     -y          - Invert y coordinate in output
    
     -n[v]       - No layout mode 'v' (=1)
     -x          - Reduce graph
    
     -Lg         - Don't use grid
     -LO         - Use old attractive force
     -Ln<i>      - Set number of iterations to i
     -LU<i>      - Set unscaled factor to i
     -LC<v>      - Set overlap expansion factor to v
     -LT[*]<v>   - Set temperature (temperature factor) to v
    
     -m          - Memory test (Observe no growth with top. Kill when done.)
     -m[v]       - Memory test - v iterations.
    
     -c          - Configure plugins (Writes $prefix/lib/graphviz/config
                   with available plugin information.  Needs write privilege.)
     -?          - Print usage and exit

...where "dot files" are plain text documents which contain [ALGOL-like](https://en.wikipedia.org/wiki/ALGOL) [syntax](https://www.graphviz.org/doc/info/lang.html) (and sometimes an [embedded SGML-like syntax](https://www.graphviz.org/doc/info/shapes.html#html)).

**Comment by ubnt-intrepid** [comment]
Thanks for report the problem.
This project was originally started for my personal configuration management and its naming did not consider the environment with Graphviz.  I agree that the project name needs to be changed, so that users who uses the environment with Graphviz's `dot` allows to use this tool.

I currently don't have any idea for a new project name, but I'd like to change it as soon as it is decided (the renaming will do at the same time as the CI migration).

Note: since `dotman` already exists [on crates.io](https://crates.io/crates/dotman), we need to choose a different name for package publishing in the future.

**Comment by daveloyall** [comment]
`dotrs` or `dotters`?  (rs for Rust.)

**Comment by marcfrederick** [comment]
I don't think including the language (Rust) in the name is a good idea, since the choice of language does not really matter or differentiate this project from others.

**Comment by waxlamp** [comment]
How about `dotmatrix`?

**Comment by sbanwart** [comment]
Looks like `dotmgr` and `dotfm` are available on [crates.io](https://crates.io).

**Comment by pickfire** [comment]
How about `tod`?

**Comment by mborejdo** [comment]
`dort` (also german, meaning "there", much better than `tod`, which means dead :-))

**Comment by pickfire** [comment]
Or it could be `dian` which means dot in chinese.

**Comment by liketechnik** [comment]
In relation to https://github.com/void-linux/void-packages/pull/20679#issuecomment-609459405: Is there a timescale planned for the rename? It would be nice to not have to rename the package after adding later on, so it would be helpful to get get some more info.

**Comment by Techcable** [comment]
One of the things I like about this project is that it's based on symbolic links.

I propose the name `linkman`: A <ins>man</ins>ager for dotfile <ins>link</ins>s 😄

I reserved it on crates.io 😉

**Comment by ubnt-intrepid** [comment]
I've rename the name of this repository to `dot.rs` to prevent GitHub's dependency analysis from mistaking it for the `dot` published on crates.io.

The change of package name is still unplanned.
